File: leetcode_python/connectedCell.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
    static int cellCounter = 0;
    static int[][] actionCosts = {{1,0},{-1,0},{0,1},{0,-1},{1,1},{1,-1},{-1,1},{-1,-1}};
    
    public static int getBiggestRegion(int[][] matrix, int n, int m) {
        int[][] visited = new int[n][m];
        int maxNum = 0;
        for(int i = 0; i < n; i++){
            for(int j = 0; j < m; j++){
                dfs(matrix,visited,i,j,n,m);
                if(cellCounter > maxNum) maxNum = cellCounter;
                cellCounter = 0;
            }
        }
        return maxNum;
    }
    
    private static void dfs(int[][] matrix, int[][] visited, int i, int j, int n, int m){
        if(i < 0 || i >= n || j < 0 || j >= m) return;
        if(visited[i][j] == 1 || matrix[i][j] == 0) return;
        cellCounter++;
        visited[i][j] = 1;
        for(int di = 0; di < 8; di++){
            dfs(matrix,visited,i+actionCosts[di][0],j+actionCosts[di][1],n,m);
        }
    }
    
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int m = in.nextInt();
        int grid[][] = new int[n][m];
        for(int grid_i=0; grid_i < n; grid_i++){
            for(int grid_j=0; grid_j < m; grid_j++){
                grid[grid_i][grid_j] = in.nextInt();
            }
        }
        System.out.println(getBiggestRegion(grid,n,m));
    }
}

File: leetcode_python/connectedCell.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();    
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////
let cellCount = 0; //better not to use global variable;
function getBiggestRegion(grid){
    
    let n = grid.length;
    let m = grid[0].length;
    let maxNum = 0;
    
    let visited = new Array(n);
    for(let i = 0; i < n; i++){
        visited[i] = new Array(m).fill(false);
    }
    const actions = [[1,0],[-1,0],[0,1],[0,-1],[1,1],[1,-1],[-1,1],[-1,-1]];
    for(let i = 0; i < n; i++){
        for(let j = 0; j < m; j++){
            dfs(grid,visited,i,j,n,m,actions);
            if(cellCount > maxNum) maxNum = cellCount;
            cellCount = 0;
        }
    }
    return maxNum;
    
}

function dfs(grid, visited, i, j, n, m,actions){
    if( i < 0 || i >= n || j < 0 || j >= m) return;
    if(visited[i][j] || grid[i][j] == 0) return;
    visited[i][j] = true;
    cellCount++;
    for(a of actions){
        dfs(grid,visited,i+a[0],j+a[1],n,m,actions);
    }
}

function main() {
    var n = parseInt(readLine());
    var m = parseInt(readLine());
    var grid = [];
    for(grid_i = 0; grid_i < n; grid_i++){
       grid[grid_i] = readLine().split(' ');
       grid[grid_i] = grid[grid_i].map(Number);
    }
    console.log(getBiggestRegion(grid));

}
